### DISCUSSION TITLE
How could i start an Java main Application with Glowroot

### DISCUSSION THREAD

**Discussion by pcouas** [glowroot/glowroot#1067] [Q&A]
Hi,
Currently i could view Tomcat Application with glowrootCentral and Cassandra 3 On windows 10, That OK
But i tried too to run an Java Batch Application and that's not running why ?

Java main class is com.zz.batch.AsyncApp
public static void main(String[] args) {

-Dglowroot.agent.id=Batch -Dglowroot.collector.address=localhost:8181 -javaagent:C:\outils\glowroot\glowroot.jar
Or
-Dglowroot.agent.id=Batch -Dglowroot.collector.address=localhost:8181 -javaagent:C:/outils/glowroot/glowroot.jar

Nothing work, could start my application but no glowroot ?
Could you help me
Thanks

**Comment by gareth-hughes-cloudpay** [comment]
Maybe this will help:
I needed to profile and upgrade a legacy command line piece of java code that ran a batch of work processing data. It takes 10 minutes.
I simply ran it in debug in Eclipse, with glowroot in the command. Just like you have.
I put a break point right at the end of the work.
Then I went to the usual glowroot url, which was working just fine, then added a custom Instrumentation to capture the workload.
When I ran it again I was able to wait for the break point, flip to glowroot and analyse the code for as long as I liked.
Glowroot never ceases to amaze me

**Reply by gareth-hughes-cloudpay** [reply]
and I simply used the -javaagent:c:/path/to/glowroot.jar
I'm on Windows, and use the forward slash
